cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/graphqlHelper.js

```diff
@@ -20,14 +20,14 @@ import { GLOBALS } from 'services/global-constants';
 function getProgram(pipeline) {
   const programs = objectQuery(pipeline, 'applicationDetail', 'programs');
 
-  if (programs === null || programs === undefined) {
-    return [];
+  if (!programs) {
+    return {};
```

Comment:
Any specific reason `programs` returning `{}` instead of `[]`?

Reply:
Because this function returns A program, not a list of programs.

Reply:
Hhmm should this variable then be `program`?

Reply:
No, because we first get all programs and then we get the program with the name based on the type of artifact.